app/src/components/DetailsBoard.js

```diff
@@ -89,16 +97,192 @@ function DetailsBoard({
+  function getSumFreeTime() {
+    //console.debug("getSumFreeTime()"); //Test
+
+    function getMilliseconds(total, current) {
+      let addedTime = current.totalTime < 0 ? 0 : current.totalTime;
+      return total + addedTime;
+    }
+
+    let msHours = 1000 * 60 * 60;
+    let msMinutes = 1000 * 60;
+
+    let totalMS = freeTimesRemaining.reduce(getMilliseconds, 0);
+    let hours = totalMS / msHours;
+    let minutes = (totalMS % msHours) / msMinutes;
+
+    setTotalFreeTime({ totalMS: totalMS, hours: hours, minutes: minutes });
+  }
+
+  //get proportion of each free time block
+  function getFreeTimePercentages() {
+    let update = freeTimesRemaining.map((freetime) => {
+      freetime.freeTimePercentage =
+        parseInt(freetime.totalTime) / totalFreeTime.totalMS;
+      return freetime;
+    });
+
+    setFreeTimesRemaining(update);
+
+    getSuggestions();
+  }
+
+  //get suggested time spent working on task
+  function getSuggestions() {
+    let task_hoursToMS = testTask.estTime.hours * 60 * 60 * 1000;
+    let msHours = 1000 * 60 * 60;
+    let msMinutes = 1000 * 60;
+
+    let totals = freeTimesRemaining.map((time) => {
+      let amount = time.freeTimePercentage * task_hoursToMS;
+      let hours = Math.round(amount / msHours);
+      let minutes = Math.round((amount % msHours) / msMinutes);
+
+      return { totalMS: amount, hours, minutes };
+    });
+
+    setSuggestions(totals);
+  }
+
+  //USE EFFECTS
+  //When component mounts, gets array of free timeblocks and total remaining time until due date
+  React.useEffect(async () => {
+    getRemainingFreeTimes(testTask.dueDate, currentDate);
+    setTimeRemaining(getRemainingTime(testTask.dueDate, currentDate));
+  }, []);
+
+  //When length of freeTimesRemaining changes (aka they've loaded in)
+  //Gets sum of all free time
+  React.useEffect(() => {
+    if (freeTimesRemaining.length !== 0) {
+      try {
+        getSumFreeTime();
+      } catch (err) {
+        console.debug("Error: ", err);
+      }
+    }
+  }, [freeTimesRemaining.length]);
+
+  //when total free time is calculated, calculates percentages
+  React.useEffect(() => {
+    if (totalFreeTime !== {}) {
+      try {
+        getFreeTimePercentages();
+      } catch (err) {
+        console.debug("Error: ", err);
+      }
+    }
+  }, [totalFreeTime]);
+
+  return (
+    <div className="calibrator">
+      <h4>Suggestions:</h4>
+      <ol>
+        {suggestions.map((time, index) => {
+          let freeTime = dayjs(freeTimesRemaining[index].start.dateTime).format(
+            "ddd, MMM D, YYYY h:mm A",
+          );
+          let duration =
+            time.totalMS >= 0
+              ? `For: ${time.hours}hrs : ${time.minutes}min`
+              : "Don't work at this time.";
+          return (
+            <li>
+              <span className="when">{`When: ${freeTime}`}</span>
+              <span className="duration">{`${duration}`}</span>
+            </li>
+          );
+        })}
```

Comment:
I might recommend extracting a `Suggestion` component here:

```
// the above `return` statement becomes something like:
  return (
    <div className="calibrator">
      <h4>Suggestions:</h4>
      <ol>
        { suggestions.map((time, index) => {
          return <Suggestion { ...{ time, suggestionIndex: index }} />
        }
      </ol>
    </div>
  );

// then, the Suggestion component looks something like:

const Suggestion = ({ suggestionIndex, time }) => {
  let freeTime = dayjs(freeTimesRemaining[suggestionIndex].start.dateTime).format(
    "ddd, MMM D, YYYY h:mm A",
  );

  const duration = time.totalMS >= 0 ? 
    `For: ${time.hours}hrs : ${time.minutes}min` : 
    "Don't work at this time.";

  return (
    <li>
      <span className="when">{`When: ${freeTime}`}</span>
      <span className="duration">{`${duration}`}</span>
    </li>
  );
}


```

Reply:
Done in next branch, haha.